pkg/server/plugin/nodeattestor/k8ssat/sat_test.go

```diff
@@ -359,26 +359,26 @@ func (s *AttestorSuite) TestConfigure() {
 func (s *AttestorSuite) TestServiceAccountKeyFileAlternateEncodings() {
 	fooPKCS1KeyPath := filepath.Join(s.dir, "foo-pkcs1.pem")
 	fooPKCS1Bytes := x509.MarshalPKCS1PublicKey(&s.fooKey.PublicKey)
-	s.Require().NoError(os.WriteFile(fooPKCS1KeyPath, pem.EncodeToMemory(&pem.Block{
+	s.Require().NoError(diskutil.WritePubliclyReadableFile(fooPKCS1KeyPath, pem.EncodeToMemory(&pem.Block{
 		Type:  "RSA PUBLIC KEY",
 		Bytes: fooPKCS1Bytes,
-	}), 0600))
+	})))
 
 	fooPKIXKeyPath := filepath.Join(s.dir, "foo-pkix.pem")
 	fooPKIXBytes, err := x509.MarshalPKIXPublicKey(s.fooKey.Public())
 	s.Require().NoError(err)
-	s.Require().NoError(os.WriteFile(fooPKIXKeyPath, pem.EncodeToMemory(&pem.Block{
+	s.Require().NoError(diskutil.WritePubliclyReadableFile(fooPKIXKeyPath, pem.EncodeToMemory(&pem.Block{
 		Type:  "PUBLIC KEY",
 		Bytes: fooPKIXBytes,
-	}), 0600))
+	})))
```

Comment:
@amartinezfayo shouldn't we use **_WritePrivateFile_** instead of **_WritePubliclyReadableFile_** here?

Reply:
I've been back and forth thinking if the calls in the testing code should be replaced. I've just found that this is problematic in the integration tests so I'll go ahead and just use the new functions in the specific places that we need to honor the permissions when running SPIRE.